src/Blockcore.Indexer.Core/Storage/Mongo/MongoStorageOperations.cs

```diff
@@ -126,60 +135,91 @@ public SyncBlockInfo PushStorageBatch(StorageBatch storageBatch)
             }
          }
 
-         var t1 = Task.Run(() =>
-         {
-            if (storageBatch.BlockTable.Values.Any())
-            data.BlockTable.InsertMany(storageBatch.BlockTable.Values, new InsertManyOptions {IsOrdered = false });
-         });
+         var blockTableTask = storageBatch.BlockTable.Values.Any()
+            ? data.BlockTable.InsertManyAsync(storageBatch.BlockTable.Values,
+               new InsertManyOptions { IsOrdered = false })
+            : Task.CompletedTask;
 
-         var t2 = Task.Run(() =>
-         {
-            if (storageBatch.TransactionBlockTable.Any())
-            data.TransactionBlockTable.InsertMany(storageBatch.TransactionBlockTable, new InsertManyOptions { IsOrdered = false });
-         });
+         var transactionBlockTableTask = storageBatch.TransactionBlockTable.Any()
+            ? data.TransactionBlockTable.InsertManyAsync(storageBatch.TransactionBlockTable,
+               new InsertManyOptions { IsOrdered = false })
+            : Task.CompletedTask;
+
+         var outputTableTask = storageBatch.OutputTable.Any()
+            ? data.OutputTable.InsertManyAsync(storageBatch.OutputTable.Values,
+               new InsertManyOptions { IsOrdered = false })
+            : Task.CompletedTask;
 
-         var t3 = Task.Run(() =>
-         {
-            if (storageBatch.OutputTable.Any())
-            data.OutputTable.InsertMany(storageBatch.OutputTable, new InsertManyOptions {IsOrdered = false});
-         });
 
-         var t4 = Task.Run(() =>
+         var inputTableTask = Task.CompletedTask;
+         if (storageBatch.InputTable.Any())
          {
-            if (storageBatch.InputTable.Any())
-               data.InputTable.InsertMany(storageBatch.InputTable, new InsertManyOptions {IsOrdered = false});
-         });
+            var utxosLookups = FetchUtxos(storageBatch.InputTable
+               .Where(_ => _.Address == null)
```

Comment:
I wonder couldn't this could be filtered in the mongo call (just an idea for improvement)